ledger/src/shred.rs

```diff
@@ -777,7 +774,23 @@ pub mod layout {
             } => {
                 merkle::ShredData::get_retransmitter_signature_offset(proof_size, chained, resigned)
             }
-        }?;
+        }
+    }
+
+    pub fn get_retransmiter_signature(shred: &[u8]) -> Result<Signature, Error> {
+        let offset = get_retransmitter_signature_offset(shred)?;
+        shred
+            .get(offset..offset + SIZE_OF_SIGNATURE)
+            .map(|bytes| <[u8; SIZE_OF_SIGNATURE]>::try_from(bytes).unwrap())
```

Comment:
`unwrap` here makes me nervous. Maybe add something like:
`.map_err(|_| Error::InvalidSignatureLength))?`

Reply:
I'm also now wondering if we have tests where we pass in invalid buffer sizes and signatures

Reply:
The very line before it:
```rust
.get(offset..offset + SIZE_OF_SIGNATURE)
```
ensures that we have a slice of size `SIZE_OF_SIGNATURE` bytes, so
```rust
<[u8; SIZE_OF_SIGNATURE]>::try_from
```
should always succeed.